mlflow/store/artifact/models_artifact_repo.py

```diff
@@ -140,16 +144,22 @@ def list_artifacts(self, path):
         return self.repo.list_artifacts(path)
 
     def _add_registered_model_meta_file(self, model_path):
-        write_yaml(
-            model_path,
-            REGISTERED_MODEL_META_FILE_NAME,
-            {
-                "model_name": self.model_name,
-                "model_version": self.model_version,
-            },
-            overwrite=True,
-            ensure_yaml_extension=False,
-        )
+        if os.path.isdir(model_path):
```

Comment:
Is this file created when we download a subdirectory in the model directory?

```
- model
    - subdir <- this directory
    - MLmodel
    - ...
```

Reply:
Could we update 'download_artifacts' function to only call this method when 'model_path' is a directory instead?

Reply:
@harupy good call. Adjusted the logic to only create this registered metadata file iff the model_path resolution after download contains the MLmodel file. Add a test to verify this behavior.

@serena-ruan great idea - updated!